### DISCUSSION TITLE
Service selection specific for each REST call based on business requirement

### DISCUSSION THREAD

**Discussion by marko-lazic** [smallrye/smallrye-stork#419] [Q&A] [answered]
The video example about Stork with heroes and villains called [Service Discovery with SmallRye Stork and Quarkus](https://www.youtube.com/watch?v=fCNwxPDGt7Q) was very good. Thank you very much for making it.

My question is about `favorite` variable and the `quarkus.stork.rest-heroes.load-balancer.favorite` property. Can I change the `favorite` programmatically in a way that it changes the service address of the one REST client API call? By that, I mean together with the specific call and not the other calls that may occur at the same time, which may have another favorite selected. So one REST Client call and one service selection should be together.

Imagine a **fake** example

```java
@GET
@Path("/random")
Uni‹Hero> findRandomHero(@Selector String favorite);

```

Is something like this possible to do or implement with Stork and is it a good idea to use it to select servers based on business requirements? I may in the distant future need multiple instances of heroes-starwars and load balancing but for now, I only need a way to call these APIs on different URL addresses. I actually want the request to fail if there are 0 services available at a particular address. What's your take on this? How would you approach this?

**Comment by cescoffier** [comment, marked as answer]
The service selection has access to the request scope data, so, you can achieve something similar by adding to the request scope the information you need to drive your selection.

**Reply by akil-rails** [reply]
Extending the example - how do I handle the case where I include fault metrics ( such as error rates in addition to least response time ) to determine the target service. ?
